addons/purchase/models/purchase.py

```diff
@@ -194,6 +194,11 @@ def name_get(self):
             result.append((po.id, name))
         return result
 
+    @api.onchange('currency_id')
+    def onchange_currency_id(self):
```

Comment:
It is mandatory to add this one? Because the currency in line is related then it should call the onchange for each lines, no ?